src/main/java/org/opentestsystem/ap/imrt/iis/service/ProjectLockServiceJdbcImpl.java

```
@@ -0,0 +1,104 @@
+                }
+                int rows = jdbcTemplate.update(UPDATE_SQL, new Object[]{now, now, projectId, existingLock});
+                if (1 != rows) {
+                    operationalEventService.serviceError(logger, null,"Unable to update project_lock entry for projectId {}", projectId);
+                    return null;
+                }
+                return now.getTime();
+            } else {
+                logger.debug("ProjectId {} is already locked", projectId);
+                return null;
+            }
+        } catch (Exception e) {
+            // The entry does not exist, create it and return the lock time
+            logger.debug("First time creating lock for project {} lock_time {}", projectId, now);
+            // If someone else got in since we checked, and inserted a row here, this will throw
+            // an exception because of the duplicate key
+            try {
+                int rows = jdbcTemplate.update(INSERT_SQL, new Object[]{projectId, now, now, now});
+                if (1 != rows) {
+                    operationalEventService.serviceError(logger, null, "Unable to create project_lock entry for projectId {}", projectId);
+                    return null;
+                }
+                return now.getTime();
+            } catch (Exception ex) {
+                return null;
+            }
+        }
+    }
+
+    @Override
+    public void unlockProject(int projectId, long lockId) {
+        Date now = new Date();
+
+        // Update the row in the database to set the lock_time back to zero, but only if the existing lock_time
+        // matches the supplied lock value
+        int rows = jdbcTemplate.update(UPDATE_SQL, new Object[]{new Date(0), now, projectId, new Date(lockId)});
+        if (1 != rows) {
+            operationalEventService.serviceError(logger, null,"Unable to unlock project for projectId {} lock {}", projectId, lockId);
```

Should this service log or should something at a higher level log?  What would the calling service do if it was unable to secure the lock?  I feel like we should throw here (runtime) rather than log allowing the client to handle the Exception or let it bubble and be caught by the general Exception handler.